xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/menus_content.vm

```diff
@@ -400,48 +408,50 @@
   ## Display all the extensions points, including the static ones
   ##
   #define ($menuContent)
-    #set ($needsSeparator = false)
     ## Admin actions UIXP
     #if ($displayAdminMenu)
       #set ($adminActions = "#displayAdminActions()")
       #if ($stringtool.isNotBlank("$!adminActions"))
         #submenuheader('core.menu.actions.main')
-        $adminActions
-        #set ($needsSeparator = true)
+        <dd>
+          <ul>
+            $adminActions
+          </ul>
+        </dd>
       #end
     #end
     ## More actions UIXP
     #if ($displayMoreActionsMenu)
       #set ($moreActions = "#displayMoreActions()")
       #if ($stringtool.isNotBlank("$!moreActions"))
-        #if ($needsSeparator)
-          #submenuseparator()
-        #end
         #submenuheader('core.menu.actions.others')
-        $moreActions
-        #set ($needsSeparator = true)
+        <dd>
+          <ul>
+            $moreActions
+          </ul>
+        </dd>
       #end
       ## ----------------------------
       ## Display the shortcuts
       ## ----------------------------
       #if($xwiki.hasAccessLevel('view') && $displayShortcuts)
         #set ($viewersContent = "#template('shortcuts.vm')")
         ## Only display a separator if at least one menu entry has been displayed before
-        #if ($needsSeparator)
-          #submenuseparator()
-        #end
         #submenuheader('core.menu.actions.viewers')
         ##
         ## VIEW SOURCE
         ##
         #if ("$!request.rev" != '')
           #set ($revisionParameter = "&rev=$request.rev")
         #end
-        #submenuitem($doc.getURL('view', "viewer=code$!{revisionParameter}") $services.localization.render('core.menu.view.source') 'tmViewSource', '', 'search', $extraAttributes)
-        $viewersContent
-        ## Display Viewers menu UIX..
-        #displaySecureUIX('org.xwiki.platform.template.menu.viewers', [])
-        #set ($needsSeparator = true)
+        <dd>
+          <ul>
+            #submenuitem($doc.getURL('view', "viewer=code$!{revisionParameter}") $services.localization.render('core.menu.view.source') 'tmViewSource', '', 'search', $extraAttributes)
+            $viewersContent
+            ## Display Viewers menu UIX..
+            #displaySecureUIX('org.xwiki.platform.template.menu.viewers', [])
```

Comment:
I've just had a look at this macro's code and unfortunately, it seems like this code could call `#submenuseparator()`. Is this a problem with these changes or would this still work?

Reply:
I'm not sure it is good to just remove the separator logic. Are you sure this isn't needed? Also, what happens if a UIX just calls `#submenuheader` (which I assume is the idea when using a separator)? Won't this break the HTML? On the other hand, https://www.xwiki.org/xwiki/bin/view/Documentation/DevGuide/ExtensionPoint/Viewers%20Menu/ doesn't even mention the parameter. However, this very same macro is used in https://www.xwiki.org/xwiki/bin/view/Documentation/DevGuide/ExtensionPoint/Edit%20Actions where it matters I think.